src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js

```diff
@@ -500,23 +500,27 @@ angular.module("umbraco")
 
             function getChildren(id) {
                 vm.loading = true;
-                return entityResource.getChildren(id, "Media", vm.searchOptions).then(function (data) {
+                return entityResource.getPagedChildren(id, "Media", vm.searchOptions).then(function (data) {
 
                     var allowedTypes = dialogOptions.filter ? dialogOptions.filter.split(",") : null;
 
-                    for (var i = 0; i < data.length; i++) {
-                        if (data[i].metaData.MediaPath !== null) {
-                            data[i].thumbnail = mediaHelper.resolveFileFromEntity(data[i], true);
-                            data[i].image = mediaHelper.resolveFileFromEntity(data[i], false);
-                        }
-                        if (data[i].metaData.UpdateDate !== null){
-                            data[i].updateDate = data[i].metaData.UpdateDate;
-                        }
-                        data[i].filtered = allowedTypes && allowedTypes.indexOf(data[i].metaData.ContentTypeAlias) < 0;
+                    // update image data to work with image grid
+                    if (data.items) {
+                        data.items.forEach(mediaItem => setMediaMetaData(mediaItem));
+                        data.items.forEach(mediaItem => mediaItem.filtered = allowedTypes && allowedTypes.indexOf(mediaItem.metaData.ContentTypeAlias) < 0);
```

Comment:
Why do we need to loop items twice?
Can't we do the following?

```
data.items.forEach(mediaItem => {
     setMediaMetaData(mediaItem);

    mediaItem.filtered = allowedTypes && allowedTypes.indexOf(mediaItem.metaData.ContentTypeAlias) < 0;
});
                    
```

Reply:
Yes